src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/AdditionalValidation.scala

```diff
@@ -30,20 +39,55 @@ trait AdditionalValidation[F[_]] {
     * @param value  the resource value
     * @return a Left(rejection) when the validation does not pass or Right(value) when it does on the effect type ''F''
     */
-  def apply(id: ResId,
-            schema: Ref,
-            types: Set[AbsoluteIri],
-            value: ResourceF.Value): EitherT[F, Rejection, ResourceF.Value]
+  def apply(id: ResId, schema: Ref, types: Set[AbsoluteIri], value: Value): EitherT[F, Rejection, Value]
 }
 
 object AdditionalValidation {
 
+  type Value = ResourceF.Value
+
   /**
     * @tparam F the monadic effect type
     * @return a new validation that always returns Right(value) on the provided effect type
     */
   final def pass[F[_]: Applicative]: AdditionalValidation[F] =
-    (id: ResId, schema: Ref, types: Set[AbsoluteIri], value: ResourceF.Value) => EitherT.rightT(value)
+    (id: ResId, schema: Ref, types: Set[AbsoluteIri], value: Value) => EitherT.rightT(value)
+
+  /**
+    * Additional validation used for checking the correctness of the ElasticSearch mappings
+    *
+    * @tparam F the monadic effect type
+    * @return a new validation that passes whenever the provided mappings are compliant with the ElasticSearch mappings or
+    *         when the view is not an ElasticView
+    */
+  final def view[F[_]](implicit F: MonadError[F, Throwable], elastic: ElasticClient[F]): AdditionalValidation[F] =
```

Comment:
I'd recommend instantiating a View using it's companion apply method, pattern match on the type and for `ElasticView` just create the index based on its `name` method. On failure -> reject, on success -> pass the validation.